### DISCUSSION TITLE
Kafka exporter restarts

### DISCUSSION THREAD

**Discussion by saloni-2404** [strimzi/strimzi-kafka-operator#5489] [General]
<html>
<body>
<div style="box-sizing: border-box; font-family: &quot;Segoe UI&quot;, system-ui, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px;">

Topic name | Partitions | Replica | consumers | messages per sec | Avg message size in KB | retention
-- | -- | -- | -- | -- | -- | --
topic1 | 200 | 3 | 80 | 30 | 17 | 24 hr
topic2 | 200 | 3 | 80 | 30 | 17 | 24 hr
topic3 | 200 | 3 | 5 | 1 | 7 | 24 hr
topic4 | 64 | 3 | 5 | 1 | 7 | 24 hr
topic5 | 64 | 3 | 10 | 3 | 10 | 24 hr

</div>
</body>
</html>

We have the above topics defined and we find the kafka exporter continuously restarting/crashing.
Do we have any limits on the number of partitions that the topic should have in order for the exporter to be able to handle the load?

**Comment by scholzj** [comment]
There are no documented limits. There were also many bugfixes in Strimzi 0.25 - so if you don't you that version, you should give it a try.

**Reply by saloni-2404** [reply]
Hey @scholzj, is there any way to configure multiple kafka exporters for a single kafka cluster?

**Reply by sknot-rh** [reply]
I do not think there is a way how to deploy multiple exporters watching a single kafka cluster. You can do it manually though.
However, I would recommend using Strimzi >=0.25 what uses the Kafka Exporter with improved stability.

**Reply by saloni-2404** [reply]
Hey @sknot-rh, thank you for responding, could you please confirm if this is what is meant by manually deploying multiple kafka exporters:
https://github.com/danielqsj/kafka_exporter

Shall we apply the file deploy/base/deployment.yaml? There is no mention of topicRegex or groupRegex there.

**Reply by sknot-rh** [reply]
Well, that looks like some basic deployment. You can do reverse engineering, let the Strimzi deploy Kafka Exporter for you, then copy the generated `Deployment`, adjust it to your needs (you can set even the number of replicas), and apply it.
```
kubectl get deployment -n your-namespace my-cluster-kafka-exporter -o yaml > custom-ke-deployment.yaml
# do the changes in the custom-ke-deployment.yaml
kubectl apply -f custom-ke-deployment.yaml -n your-namespace
```
Attaching an example deployment.
[custom-kafka-exporter.yaml.txt](https://github.com/strimzi/strimzi-kafka-operator/files/7202405/custom-kafka-exporter.yaml.txt)
